openmc/deplete/helpers.py

```diff
@@ -142,3 +143,195 @@ def update(self, fission_rates, _mat_index):
+
+        # Tally energy-weighted group-wise fission reaction rate
+        # Used to evaluated linear interpolation between fission yield points
+        self._weighted_reaction_tally = Tally()
+        self._weighted_reaction_tally.scores = ['fission']
+
+        filters = [
+            MaterialFilter(materials), EnergyFilter(self._energy_bounds)]
+
+        self._reaction_tally.filters = filters
+        self.local_indexes = asarray(mat_indexes)
+
+    def set_fissionable_nuclides(self, nuclides):
+        """List of string of nuclides with data to be tallied
+
+        Parameters
+        ----------
+        nuclides : iterable of str
+            Nuclides with non-zero densities that are candidates
+            for the fission tally. Not necessary that all are nuclides
+            with fission yields, but at least one must be
+
+        Returns
+        -------
+        nuclides : tuple of str
+            Nuclides ordered as they appear in the tally and in
+            the nuclide column of :attr:`results`
+
+        Raises
+        ------
+        ValueError
+            If no nuclides in ``nuclides`` are tracked on this
+            object
+        """
+        # Set of all nuclides with positive density
+        # and fission yield data
+        nuc_set = self._chain_set & set(nuclides)
+        if len(nuc_set) == 0:
+            raise ValueError(
+                "No overlap between chain nuclides with fission yields and "
+                "requested tally nuclides")
+        nuclides = tuple(sorted(nuc_set))
+        self._tally_index = [self._chain_nuclides[n] for n in nuclides]
+        self._reaction_tally.nuclides = nuclides
+        return nuclides
+
+    def unpack(self):
+        """Unpack fission rate tallies to produce :attr:`results`
+        """
+        # if this process is not responsible for depleting anything
+        # [more processes than burnable materials]
+        # don't do anything
+        if self.local_indexes.size == 0:
+            return
+
+        # get view into tally results
+        # new shape: [material, energy, parent nuclide]
+        result_view = self._reaction_tally.results[..., 1].reshape(
+            self.n_bmats, len(self._energy_bounds) - 1,
+            len(self._reaction_tally.nuclides))
+
+        # Get results specific to this process
+        fission_rates = result_view[self.local_indexes]
+        self.results = empty_like(fission_rates)
+
+        # scale group fission rates proportional to total fission rate
+        fission_total = fission_rates.sum(axis=1)
+        nz_mat, nz_nuc = fission_total.nonzero()
+        self.results[nz_mat, :, nz_nuc] = (
+            fission_rates[nz_mat, :, nz_nuc]
+            / fission_total[nz_mat, newaxis, nz_nuc])
+
+        # directly set values to zero where total fission rate is zero
+        z_mat, z_nuc = where(fission_total == 0.0)
+        self.results[z_mat, :, z_nuc] = 0.0
+
+    def compute_yields(self, local_mat_index):
```

Comment:
I would call this `weighted_yields` 1) to indicate that there is a return value and 2) to give insight into what it's doing